neurodiffeq/solvers_utils.py

```diff
@@ -530,7 +530,9 @@ def load(cls,
                          t_min=t_min,
                          t_max=t_max,
                          theta_min=tuple(load_dict['solver'].r_min[1:]),
-                         theta_max=tuple(load_dict['solver'].r_max[1:]))
+                         theta_max=tuple(load_dict['solver'].r_max[1:]),
+                         eq_param_index=load_dict['solver'].eq_param_index

```

Comment:
At a certain point we should consider refactoring the solver_utils package. A lot of the code seems to be a little ad-hoc. They are fine for now.

Reply:
Yes, I think we are due to think about refactoring and redesigning these pieces as they are pretty hard coded. I also have some thoughts on how parameters are saved - https://github.com/NeuroDiffGym/neurodiffeq/blob/master/neurodiffeq/solvers_utils.py#L99 which ties back to how parameters are used inside the function of the differential equation. I will open up an issue on this and we can discuss these things in detail.